SpatialGDK/Source/SpatialGDK/Private/Utils/EntityPool.cpp

```diff
@@ -21,12 +21,15 @@ void UEntityPool::Init(USpatialNetDriver* InNetDriver, FTimerManager* InTimerMan
 	ReserveEntityIDs(GetDefault<USpatialGDKSettings>()->EntityPoolInitialReservationCount);
 }
 
-void UEntityPool::ReserveEntityIDs(int32 EntitiesToReserve)
+void UEntityPool::ReserveEntityIDs(uint32 EntitiesToReserve)
 {
 	UE_LOG(LogSpatialEntityPool, Verbose, TEXT("Sending bulk entity ID Reservation Request for %d IDs"), EntitiesToReserve);
 
 	checkf(!bIsAwaitingResponse, TEXT("Trying to reserve Entity IDs while another reserve request is in flight"));
 
+	// TODO: UNR-4979 Allow full range of uint32 when SQD-1150 is fixed
+	EntitiesToReserve = FMath::Clamp(EntitiesToReserve, 0u, static_cast<uint32>(MAX_int32));
```

Comment:
Might suggest adding a log if the clamp will reduce the size of the reservation request.